java/src/com/ibm/streamsx/topology/internal/context/remote/ZippedToolkitRemoteContext.java

```diff
@@ -128,22 +131,36 @@ private static Path pack(final File appTkRoot, JsonObject submission, String tkN
         Path manifestTmp = Files.createTempFile("manifest_tk", ".txt");
         Path mainCompTmp = Files.createTempFile("main_composite", ".txt");
         Path scOptsTmp = Files.createTempFile("sc_opts", ".txt");
-        
+        Path splmmOptsTmp = Files.createTempFile("splmm_opts", ".txt");
+        Path splmm_dirTmp = Files.createTempFile("splmm_dir", ".txt");
+
         // tkManifest is the list of toolkits contained in the archive
         try (PrintWriter tkManifest = new PrintWriter(manifestTmp.toFile(), "UTF-8")) {
             if (tkName != null)
                 tkManifest.println(tkName);
             tkManifest.println(topologyToolkitName);
-            
             JsonObject configSpl = object(graph, CONFIG, "spl");
             if (configSpl != null) {
                 objectArray(configSpl, "toolkits",
                         tk -> {
                             if (tk.has("root")) {
                                 File tkRoot = new File(jstring(tk, "root"));
                                 String tkRootName = tkRoot.getName();
-                                tkManifest.println(tkRootName);
-                                toolkits.put(tkRoot.toPath(), tkRootName);
+                                // If SPLMM app, need to seperate this toolkit to run the command
+                                // 'spl-make-toolkit -i <SPLMM_APP_DIR> `cat splmm_opts.txt`'
+                                // Bc SPLMM app requires arguments, so can't group with other toolkits
+                                if (deploy(submission).has(ContextProperties._SPLMM_OPTIONS)) {
+                                    if (SPLMM_toolkits.isEmpty()) {
```

Comment:
Is there any reason to split the SPLMM toolkit into its own map, rather than just including it in `toolkits`. It seems the same code is executed for both.

Reply:
Yes, it goes back to the case I spoke about above. If I don't separate it, the makefile will error on this line `${STREAMS_INSTALL}/bin/spl-make-toolkit -i $${toolkit}; \` since the SPLMM toolkit args are missing. An alternative that worked before was to edit that line to be `${STREAMS_INSTALL}/bin/spl-make-toolkit -i $${toolkit} `cat splmm_opts.txt`; \` but I wasn't confident that the SPLMM opts would not interfere with the other toolkits.

Reply:
Isn't that command driven by `tkManifest`, not `toolkits`?

Isn't `toolkits` (from line 125) just used as the list of toolkits to copy into the code archive, which is exactly what `SPLMM_toolkits` (line 121) is doing?

Reply:
I see what you're saying. I will try that and re-run the tests

Reply:
Removed SPLMM toolkit map and included SPLMM toolkit in toolkits map, in bcb9110